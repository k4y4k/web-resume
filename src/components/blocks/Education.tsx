import 'twin.macro'
import * as React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import ItemContainer from '../section/ItemContainer'
import SectionContainer from '../section/SectionContainer'

interface EducationItem {
  courses: string[]
  institution: string
  area: string
  studyType: string
  startDate: string
  endDate: string
}

interface two {
  history: EducationItem[]
}

export const PureEducation = ({ history }: two): JSX.Element => {
  return (
    <div data-testid='education'>
      <SectionContainer title='Education'>
        {history.map((el, i: number) => {
          return (
            <ItemContainer
              key={i}
              courses={el.courses}
              institution={el.institution}
              area={el.area}
              studyType={el.studyType}
              fromDate={el.startDate}
              toDate={el.endDate}
            />
          )
        })}
      </SectionContainer>
    </div>
  )
}

export const Education = (): JSX.Element => {
  const data = useStaticQuery(graphql`
    {
      file(name: { eq: "data" }, extension: { eq: "json" }) {
        childDataJson {
          education {
            institution
            area
            studyType
            startDate
            endDate
            courses
          }
        }
      }
    }
  `)

  const { education } = data.file.childDataJson

  return <PureEducation history={education} />
}

export default Education
