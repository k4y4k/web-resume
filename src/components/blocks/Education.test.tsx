import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { PureEducation as Education } from './Education'

describe('<Education />', () => {
  test('bails out on no data', () => {
    render(<Education history={[]} />)

    // still want the title
    expect(screen.getByTestId('education')).toBeInTheDocument()

    // if there's no data, we shouldn't see <ItemContainer /> render
    expect(screen.queryByTestId('sectionItemContainer')).not.toBeInTheDocument()
    expect(screen.getByTestId('education')).toMatchSnapshot()
  })

  test('renders array of >=1 Education items', () => {
    const data = [
      {
        institution: 'University',
        area: 'Software Development',
        studyType: 'Bachelor',
        startDate: '2011-01-01',
        endDate: '2013-01-01',
        gpa: '4.0',
        courses: ['DB1101 - Basic SQL'],
      },
    ]

    const { container } = render(<Education history={data} />)

    const EducationList = screen.getByTestId('education')

    expect(EducationList).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
