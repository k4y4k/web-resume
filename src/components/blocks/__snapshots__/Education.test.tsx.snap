// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Education /> bails out on no data 1`] = `
<div
  data-testid="education"
>
  <div
    css="[object Object],[object Object],&:hover {
	box-shadow:
	  0 2.8px 2.2px rgba(181,61,39, 0.07),
	  0 6.7px 5.3px rgba(181,61,39, 0.05),
	  0 12.5px 10px rgba(181,61,39, 0.042),
	  0 22.3px 17.9px rgba(181,61,39, 0.035),
	  0 41.8px 33.4px rgba(181,61,39, 0.028),
	  0 100px 80px rgba(181,61,39, 0.02);
	},[object Object]"
  >
    <h2
      css="[object Object],[object Object]"
    >
      Education
    </h2>
     
    <br />
     
  </div>
</div>
`;

exports[`<Education /> renders array of >=1 Education items 1`] = `
.c5 {
  display: inline-block;
}

.c3 {
  display: inline-block;
}

.c1 {
  font-size: 1.5rem;
  line-height: 2rem;
  text-transform: uppercase;
  font-weight: 900;
}

.c4 {
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  display: inline-block;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1.5rem;
  margin: 1.5rem;
  margin-top: 0px;
  margin-bottom: 0px;
  --tw-bg-opacity: 0.75;
  border-bottom-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgba(0,0,0,var(--tw-border-opacity));
  border-style: none;
}

.c2 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  font-style: italic;
}

@media print {
  .c5 {
    font-size: 0.875rem;
    line-height: 1.25rem;
  }
}

@media print {
  .c1 {
    font-size: 1.125rem;
    line-height: 1.75rem;
  }
}

@media print {
  .c0 {
    padding: 0px;
    margin: 0px;
    margin-left: 1.5rem;
    margin-right: 1.5rem;
    padding-top: 0.75rem;
    padding-bottom: 0.75rem;
    border-width: 0px;
  }
}

@media print {
  .c2 {
    margin: 0px;
  }
}

<div
  data-testid="education"
>
  <div
    css="[object Object],[object Object],&:hover {
	box-shadow:
	  0 2.8px 2.2px rgba(181,61,39, 0.07),
	  0 6.7px 5.3px rgba(181,61,39, 0.05),
	  0 12.5px 10px rgba(181,61,39, 0.042),
	  0 22.3px 17.9px rgba(181,61,39, 0.035),
	  0 41.8px 33.4px rgba(181,61,39, 0.028),
	  0 100px 80px rgba(181,61,39, 0.02);
	},[object Object]"
  >
    <h2
      css="[object Object],[object Object]"
    >
      Education
    </h2>
     
    <br />
     
    <div
      class="c0"
      data-testid="sectionItemContainer"
    >
      <h1
        class="c1"
      >
        Bachelor of Software Development
      </h1>
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          University
        </h2>
        <div
          class="c4"
        >
          |
        </div>
        <p
          class="c5"
          data-testid="sectionItemDates"
        >
          2011
           - 
          2013
        </p>
      </div>
      <div
        css="[object Object],li {list-style-type: circle; list-style-position: inside;}"
        data-testid="sectionItemDetails"
      >
        <ul>
          <li>
            DB1101 - Basic SQL
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
